src/docs/guides/flask.md

```diff
@@ -0,0 +1,221 @@
+
+## Deploy from the CLI
+
+1. **Install the Railway CLI**:
+    - <a href="/guides/cli#installing-the-cli" target="_blank">Install the CLI</a> and <a href="/guides/cli#authenticating-with-the-cli" target="_blank">authenticate it</a> using your Railway account.
+2. **Initialize a Railway Project**:
+    - Run the command below in your Flask app directory. 
+        ```bash
+        railway init
+        ```
+    - Follow the prompts to name your project.
+    - After the project is created, click the provided link to view it in your browser.
+3. **Deploy the Application**:
+    - Use the command below to deploy your app:
+        ```bash
+        railway up
+        ```
+    - This command will scan, compress and upload your app's files to Railway. You’ll see real-time deployment logs in your terminal.
+4. **Verify the Deployment**:
+    - Once the deployment completes, go to [**View logs**](/guides/logs#build--deploy-panel) to check if the server is running successfully.
+6. **Set Up a Public URL**:
+    - Navigate to the **Networking** section under the [Settings](/overview/the-basics#service-settings) tab of your new service.
+    - Click [Generate Domain](/guides/public-networking#railway-provided-domain) to create a public URL for your app.
+
+<Image src="https://res.cloudinary.com/railway/image/upload/f_auto,q_auto/v1730473731/docs/quick-start/flask_app_on_railway.png"
+alt="screenshot of the deployed Flask service"
+layout="responsive"
+width={2164} height={1814} quality={100} />
+
+## Deploy from a GitHub Repo
+
+To deploy a Flask app to Railway directly from GitHub, follow the steps below:
+
+1. **Create a New Project on Railway**:
+    - Go to <a href="https://railway.app/new" target="_blank">Railway</a> to create a new project.
+2. **Deploy from GitHub**: 
+    - Select **Deploy from GitHub repo** and choose your repository.
+        - If your Railway account isn’t linked to GitHub yet, you’ll be prompted to do so.
+3. **Deploy the App Service**:
+    - Click **Deploy** on the Railway project canvas to apply your changes.
+4. **Verify the Deployment**:
+    - Once the deployment completes, go to [**View logs**](/guides/logs#build--deploy-panel) to check if the server is running successfully.
+
+    **Note:** During the deployment process, Railway will automatically [detect that it’s a Python app](https://nixpacks.com/docs/providers/python).
+5. **Set Up a Public URL**:
+    - Navigate to the **Networking** section under the [Settings](/overview/the-basics#service-settings) tab of your new service.
+    - Click [Generate Domain](/guides/public-networking#railway-provided-domain) to create a public URL for your app.
+
+## Use a Dockerfile
+
+1. Create a `Dockerfile` in the app's root directory.
+2. Add the content below to the `Dockerfile`:
+    ```docker
+    # Use the Python 3 alpine official image
+    # https://hub.docker.com/_/python
+    FROM python:3-alpine
```

Comment:
We try to shy away from alpine images, they have a strange networking stack and have shown to have lower CPU performance compared to the Debian counterparts.

Reply:
Interesting...never knew that.